tools/toolchains/arm.py

```diff
@@ -252,7 +257,8 @@ def archive(self, objects, lib_path):
     @hook_tool
     def binary(self, resources, elf, bin):
         _, fmt = splitext(bin)
-        bin_arg = {".bin": "--bin", ".hex": "--i32"}[fmt]
+        # On .hex format, combine multiple .hex files (for multiple load regions) into one 
+        bin_arg = {".bin": "--bin", ".hex": "--i32combined"}[fmt]
```

Comment:
Good catch. This was always the indented behavior. I guess nobody combined multiple load regions with hex output before.